contracts/bots/PartialLiquidationBot.sol

```diff
@@ -0,0 +1,292 @@
+    ///         Accepts the amount of underlying that the liquidator is willing to spend
+    /// @param creditManager Credit Manager where the account resides
+    /// @param creditAccount Credit Account to liquidate
+    /// @param assetOut Asset to receive for underlying
+    /// @param amountIn Amount of underlying to pay. Note that depending on the CA's balance in assetOut and
+    ///                 the maximal swappable amount of underlying, the actual spent amount may be less
+    /// @param repay Whether to repay debt or only perform a swap
+    /// @param priceUpdates Data for updatable price feeds, if any are required to compute account health
+    function partialLiquidateExactIn(
+        address creditManager,
+        address creditAccount,
+        address assetOut,
+        uint256 amountIn,
+        address to,
+        bool repay,
+        PriceUpdate[] memory priceUpdates
+    ) external returns (uint256, uint256) {
+        LiquidationParams memory params =
+            _prepareParams(creditManager, creditAccount, assetOut, amountIn, 0, to, repay, true);
+
+        /// Since we are computing debt and collateral before liquidation,
+        /// we need to update prices beforehand, if needed
+        _applyPriceUpdates(priceUpdates, params.priceOracle);
+
+        _retrieveAndCheckDebt(params);
+
+        _liquidate(params);
+
+        return (params.amountIn, params.amountOut);
+    }
+
+    /// @notice Performs a partial liquidation by swapping some CA collateral to underlying at a discount.
+    ///         Accepts the amount of assetOut that the liquidator wishes to receive
+    /// @param creditManager Credit Manager where the account resides
+    /// @param creditAccount Credit Account to liquidate
+    /// @param assetOut Asset to receive for underlying
+    /// @param amountOut Amount of assetOut to receive. Note that depending on the CA's balance in assetOut and
+    ///                 the maximal swappable amount of underlying, the actual received amount may be less
+    ///                 (with the spent amount adjusted accordingly)
+    /// @param repay Whether to repay debt or only perform a swap
+    /// @param priceUpdates Data for updatable price feeds, if any are required to compute account health
+    function partialLiquidateExactOut(
+        address creditManager,
+        address creditAccount,
+        address assetOut,
+        uint256 amountOut,
+        address to,
+        bool repay,
+        PriceUpdate[] memory priceUpdates
+    ) external returns (uint256, uint256) {
+        LiquidationParams memory params =
+            _prepareParams(creditManager, creditAccount, assetOut, 0, amountOut, to, repay, false);
+
+        /// Since we are computing debt and collateral before liquidation,
+        /// we need to update prices beforehand, if needed
+        _applyPriceUpdates(priceUpdates, params.priceOracle);
+
+        _retrieveAndCheckDebt(params);
+
+        _liquidate(params);
```

Comment:
It seems, that repay could be a parameter here

Reply:
"repay" is already in params

Reply:
I understood that it's used in this function only, proboaly, it's better to move it out from params and use directly here.

Reply:
It is also used in _getMulticall()